### PR TITLE
snowflake_role_grants resource

### PR DESCRIPTION
**Pull request by ryanking**
This resource supports granting roles to other roles and users. This grants to those entities the ability to switch to this role.

This resource is modeled after the [aws_iam_policy_attachment](https://www.terraform.io/docs/providers/aws/r/iam_policy_attachment.html) and is intended to exclusively manage the role grants for a role.

### CODE REVIEW

File: pkg/resources/role_grants.go [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+	return ReadRoleGrants(data, meta)
+}
+
+func grantRoleToRole(db *sql.DB, role1, role2 string) error {
+	g := snowflake.RoleGrant(role1)
+	_, err := db.Exec(g.Role(role2).Grant())
+	return err
+}
+
+func grantRoleToUsers(db *sql.DB, roleName string, users []string) error {
```
Comment (edulop91) [inline pkg/resources/role_grants.go]: // TODO

File: pkg/resources/role_grants.go [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+	return nil
+}
+
+func grantRoleToUser(db *sql.DB, role1, user string) error {
+	g := snowflake.RoleGrant(role1)
+	_, err := db.Exec(g.User(user).Grant())
+	return err
+}
+
+// Trying out this new way of reading rows. If it works well, will move to pkg/snowflake and apply to other
```
Comment (edulop91) [inline pkg/resources/role_grants.go]: nice

File: pkg/resources/role_grants.go (hunk 1 above)
Comment (ryanking) [reply pkg/resources/role_grants.go]: Ha good catch. Not meant to be TODO, not sure why tests didn't catch it.

Comment (ryanking) [reply pkg/resources/role_grants.go]: nm actually meant to be deleted

File: pkg/snowflake/role_grant.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
+package snowflake
+
+import "fmt"
+
+type RoleGrantBuilder struct {
```
Comment (edulop91) [inline pkg/snowflake/role_grant.go]: nice

File: pkg/resources/role_grants.go (hunk 1 above)
Comment (edulop91) [reply pkg/resources/role_grants.go]: :+1:

File: pkg/resources/role_grants.go [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+		case "ROLE":
+			roles = append(roles, grant.GranteeName.String)
+		case "USER":
+			users = append(users, grant.GranteeName.String)
+		default:
+			return fmt.Errorf("unknown grant type %s", grant.GrantedTo.String)
+		}
+	}
+
+	data.Set("roles", roles)
```
[the remark this answers is not in the document]
Comment (ryanking) [reply pkg/resources/role_grants.go]: added a test

**Pull request merged**